src/cargo/util_semver.rs

```diff
@@ -0,0 +1,195 @@
+use std::fmt::{self, Display};
```

Comment:
Why is `OptVersionReq` left out in the party? Because it is not a dependency of `schema.rs`?

Reply:
Not quite sure where it should live.  It isn't just a simple semver extension like `VersionExt` or `PartialVersion` but is tightly coupled to specific parts of cargo, like `RustVersion`.